<a href="https://colab.research.google.com/github/yovngburgos/examenEstadistica/blob/main/Laboratorio_4_Medidas_de_Tendencia_Central_y_Percentiles.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Medidas de Tendencia Central y Percentiles

## Objetivo:

Aplicar las medidas de tendencia central (media, mediana y moda) y medidas de posición no central (percentiles) en el análisis de datos, utilizando Python como herramienta de apoyo.

*Nota: Este laboratorio está diseñado para desarrollar razonamiento estadístico. Evita el uso de herramientas de IA para responder las preguntas a menos que esto sea explícitamente sugerido* 🤖. *En su lugar, enfócate en aplicar los conceptos revisados.*

## Actividad 1

Imagina que trabajas como analista de datos en una consultora de orientación vocacional que asesora a estudiantes de enseñanza media sobre qué carrera elegir. Recibes una base de datos con información de carreras técnicas y profesionales de distintas instituciones chilenas de los años 2024 y 2025. Tu misión es analizar los ingresos y empleabilidad de los titulados, y entregar conclusiones claras que ayuden a tomar decisiones.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://docs.google.com/spreadsheets/d/1JmPTeHd5mzM4V8A2yLz0Jhwt6MDmmaxf/export?format=xlsx
">aquí</a>**.

In [ ]:
import pandas as pd
pd.options.display.float_format='{:,.2f}'.format

base_datos=pd.read_excel('Actividad_1_2_3.xlsx')

In [ ]:
#Tamaño base de datos:
base_datos.shape[0]


1163

In [ ]:
base_datos.head(3)

Código           Tipo de institución  Acreditación institución (años)  \
0     456  Centros de Formación Técnica                                5   
1     456  Centros de Formación Técnica                                5   
2     456  Centros de Formación Técnica                                5   

               Área                            Nombre carrera genérica  \
0        Tecnología  Técnico en Electricidad y Electricidad Industrial   
1             Salud                              Técnico en Enfermería   
2  Ciencias Básicas                              Técnico en Geominería   

  Duración Real (semestres)  Empleabilidad 1er año (%)  \
0                      6.54                      76.62   
1                      6.28                      62.50   
2                      7.91                      61.63   

   Ingreso al 4° año (pesos)  
0                    1150000  
1                     750000  
2                    1150000

1. Identifica la población y muestra del estudio.

**Población:** carreras técnicas y profesionales de distintas instituciones chilenas de los años 2024 y 2025

**Muestra:** 1163 carreras técnicas y profesionales de distintas instituciones chilenas de los años 2024 y 2025

2. Clasifica las variables presentes en la base de datos.

Tipo de institución:Cualitativa nominal

Acreditación: Cuantitativa discreta

Área: Cualitativa nominal

Nombre Carrera: Cualitativa nominal

Duración real: Cuantitativa continua

Empleabilidad 1er año: Cuantitativa continua

Ingreso al 4to año: Cuantitativa continua

3. ¿Cuál es el ingreso promedio de los titulados? ¿Y la mediana? ¿Qué puedes concluir de la diferencia entre ambos?

In [ ]:
base_datos['Ingreso al 4° año (pesos)'].mean()

np.float64(1307480.653482373)

In [ ]:
base_datos['Ingreso al 4° año (pesos)'].median()

1150000.0

El ingreso promedio es de \$1307480 aprox. y su mediana es de \$1150000.

4. ¿Cuál es la moda de los ingresos? Interprétala.

In [ ]:
base_datos['Ingreso al 4° año (pesos)'].mode()

0    1050000
Name: Ingreso al 4° año (pesos), dtype: int64

**Resuesta:** El sueldo más frecuente es de \$1050000

5. Calcula los percentiles 25, 50 y 75 del ingreso de los titulados e interprétalos.

In [ ]:
print(f'Percentil 25={base_datos['Ingreso al 4° año (pesos)'].quantile(0.25)}')
print(f'Percentil 50={base_datos['Ingreso al 4° año (pesos)'].quantile(0.5)}')
print(f'Percentil 75={base_datos['Ingreso al 4° año (pesos)'].quantile(0.75)}')

Percentil 25=950000.0
Percentil 50=1150000.0
Percentil 75=1550000.0


**Interpretaciones:**

**Percentil 25:** El 25% de los ingresos de los titulados no superan los 950000 pesos

**Percentil 75:** El 75% de los ingresos de los titulados no superan los 1550000 pesos

**Percentil 50:** El 50% de los ingresos de los titulados no superan los 1150000 pesos

6. Utiliza la función *describe()* de Pandas para validar tus resultados.

In [ ]:
base_datos['Ingreso al 4° año (pesos)'].describe()

count       1,163.00
mean    1,307,480.65
std       506,349.46
min       550,000.00
25%       950,000.00
50%     1,150,000.00
75%     1,550,000.00
max     3,000,000.00
Name: Ingreso al 4° año (pesos), dtype: float64

7. Considera que tres personas desean estudiar Técnico en enfermería, pero cada una valora cosas distintas:
* Sofía busca maximizar su ingreso futuro.
* Diego prioriza encontrar trabajo rápidamente.
* Camila quiere un equilibrio entre ambas.

$\hspace{1cm}$Considerando la media como medida representativa, ¿qué tipo de institución recomendarías a cada uno y por qué?

In [ ]:
#Para Sofía:
base_datos[base_datos["Nombre carrera genérica"] == "Técnico en Enfermería"].groupby("Tipo de institución")["Ingreso al 4° año (pesos)"].mean()

Tipo de institución
Centros de Formación Técnica   740,909.09
Insitutos Profesionales        750,000.00
Universidades                  790,000.00
Name: Ingreso al 4° año (pesos), dtype: float64

**Respuesta:** A sofía se le recomienda ingresar a una Universidad pues su sueldo promedio es el mayor.

In [ ]:
#Para Diego:
base_datos[base_datos["Nombre carrera genérica"] == "Técnico en Enfermería"].groupby("Tipo de institución")["Empleabilidad 1er año (%)"].mean()

Tipo de institución
Centros de Formación Técnica   63.98
Insitutos Profesionales        66.85
Universidades                  65.05
Name: Empleabilidad 1er año (%), dtype: float64

**Respuesta:** A Diego le conviene un IP pues tiene mayor promedio de empleabilidad.

**Respuesta:** En el caso de Camila, elegir IP o Universidad puede ser buena opción pues en ambos casos tienen buen ingreso promedio y tambien buena empleabilidad.

## Actividad 2

Un grupo de estudiantes está evaluando si el nivel de acreditación de una institución podría impactar sus resultados laborales. Para responder esta pregunta, realiza el siguiente análisis:

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://docs.google.com/spreadsheets/d/1JmPTeHd5mzM4V8A2yLz0Jhwt6MDmmaxf/export?format=xlsx
">aquí</a>**.

1. Agrupa los datos por años de acreditación y calcula la media y la mediana del ingreso al 4to año de cada grupo.

In [ ]:
base_datos.groupby('Acreditación institución (años)')['Ingreso al 4° año (pesos)'].mean()

Acreditación institución (años)
0   1,024,666.67
3   1,313,043.48
4   1,153,020.13
5   1,192,729.59
6   1,541,260.16
7   1,421,043.17
Name: Ingreso al 4° año (pesos), dtype: float64

In [ ]:
base_datos.groupby('Acreditación institución (años)')['Ingreso al 4° año (pesos)'].median()

Acreditación institución (años)
0     950,000.00
3   1,150,000.00
4   1,050,000.00
5   1,050,000.00
6   1,450,000.00
7   1,250,000.00
Name: Ingreso al 4° año (pesos), dtype: float64

2. ¿Existe una relación aparente entre la cantidad de años de acreditación y el ingreso promedio? ¿Qué patrón observas?

Podemos destacar que con 6 años de acreditación el ingreso promedio es el mayor.
Tambien se destaca que el 3er promedio más alto es con 3 años de acreditación.

3. ¿Se observa una tendencia similar al analizar la empleabilidad al primer año?

In [ ]:
base_datos.groupby('Acreditación institución (años)')['Empleabilidad 1er año (%)'].mean()

Acreditación institución (años)
0   63.92
3   69.28
4   63.34
5   63.65
6   71.26
7   71.73
Name: Empleabilidad 1er año (%), dtype: float64

In [ ]:
base_datos.groupby('Acreditación institución (años)')['Empleabilidad 1er año (%)'].median()

Acreditación institución (años)
0   69.40
3   69.05
4   65.00
5   65.60
6   73.99
7   74.25
Name: Empleabilidad 1er año (%), dtype: float64

La tendencia es similar a lo que ocurre con el ingreso promedio y su mediana, salvo que la mayor empleabilidad (promedio y mediana) se obtiene con 7 años de acreditación. En esta variable si podemos decir que "a mayor acreditación, mayor será la emplebilidad"

4. Comparando las variables ingreso y empleabilidad, analiza cómo varían la media y la mediana en cada caso. ¿Qué conclusiones generales puedes extraer sobre la representatividad de estas medidas?

5. Teniendo en cuenta tu análisis sobre cómo varían los ingresos y la empleabilidad con los años de acreditación, ¿qué consejo le darías a un estudiante que va a elegir una institución basándose únicamente en el nivel de acreditación? Justifica tu recomendación.

## Actividad 3

Imagina que trabajas como analista del Ministerio de Educación. Te han solicitado caracterizar los resultados laborales de los titulados por área de formación utilizando una base de datos con información de carreras técnicas y profesionales de distintas instituciones chilenas de los años 2024 y 2025.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://docs.google.com/spreadsheets/d/1JmPTeHd5mzM4V8A2yLz0Jhwt6MDmmaxf/export?format=xlsx
">aquí</a>**.

1. ¿Cuál Área se observa con mayor frecuencia entre los titulados?

2. ¿Entre qué sueldos se encuentran los egresados del 10% más alto al 4to año de titulación?

3. ¿Entre qué sueldos se encuentran los egresados del 17% más bajo al 4to año de titulación?

4. Agrupa los datos por Área y para cada grupo calcula las siguientes medidas del Ingreso al 4° año:
* Media
* Mediana
* Percentil 25
* Percentil 75

$\hspace{1cm}$Luego, construye un DataFrame que reúna estas cuatro columnas.

5. Interpreta las medidas calculadas para el área de Agropecuaria y Tecnología.

6. Considerando las medidas calculadas, para el análisis del Área de Salud, ¿qué medida considerarías más representativa? Comenta con tus compañeros.

7. Compara la media del ingreso al 4to año de titulación, por Área y por Tipo de institución. ¿Qué conclusiones puedes extraer acerca del impacto del tipo de institución en el ingreso según área?

*Sugerencia: Puedes crear una tabla bivariada utilizando la función crosstab() de Pandas*.

## Actividad 4

La siguiente base de datos posee los registros de todas las calificaciones obtenidas por los alumnos del segundo semestre del 2019, jornada diurna en la asignatura de Álgebra.

Para responder las siguientes preguntas, descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1cD6GAWUzR2G8eHEEnfPHIuzKIZtHcSf7
">aquí</a>**.

1. ¿Cuál es la población en estudio?

2. ¿Cuál es la muestra en estudio?

3.  Identifique y clasifique las variables involucradas en el estudio.

4. Calcule e interprete las medidas de tendencia central de las notas del examen.

5. ¿Entre qué valores se encuentra el 60% de las notas más bajas en la prueba 1?

6.  La institución de educación creará cursos de reforzamiento para todos los estudiantes cuyas calificaciones en la Prueba 2, estén en el 25% más bajo. ¿Cuál es la nota más alta dentro de este grupo que va a reforzamiento?

7. ¿Cuál es la nota más baja entre el 30% de mejores notas del examen?

8. Si se desea reevaluar al 10% más bajo en el examen, ¿cuál es rango de notas que se debe considerar?

9. Si la institución va a premiar al 5% de las mejores notas finales, ¿cuál es la nota más baja registrada por ese grupo?